core/src/main/java/com/sequenceiq/distrox/v1/distrox/converter/DistroXV1RequestToStackV4RequestConverter.java

```diff
@@ -99,9 +100,31 @@ public StackV4Request convert(DistroXV1Request source) {
         request.setTelemetry(getTelemetryRequest(source, environment, sdxClusterResponse));
         request.setGatewayPort(source.getGatewayPort());
         request.setExternalDatabase(getIfNotNull(source.getExternalDatabase(), databaseRequestConverter::convert));
+        request.setEnableLoadBalancer(source.isEnableLoadBalancer());
+        checkMultipleGatewayNodes(source);
         return request;
     }
 
+    private void checkMultipleGatewayNodes(DistroXV1Request distroXV1Request) {
+        if (distroXV1Request.getInstanceGroups() == null || distroXV1Request.getInstanceGroups().isEmpty()) {
+            return;
+        }
+        distroXV1Request.getInstanceGroups().forEach(ig -> {
+            if (InstanceGroupType.GATEWAY == ig.getType()) {
```

Comment:
The current load balancer logic already checks for the existence of the Knox service, and will only create load balancers if Knox is defined either explicitly or implicitly. Is there a reason not related to Knox we're checking the instance type here?

Reply:
Yes, the datahub has a special check not to allow GATEWAY node type to be defined for more than one node. This change basically enables to relax that check when a load-balancer is defined. This will enable us to have a master group with two nodes as GATEWAY where in the future we will also have CM HA.